graphql_compiler/compiler/emit_sql.py

```diff
@@ -239,19 +340,152 @@ def start_global_operations(self):
+            self._ir.query_metadata_table.get_location_info(fold_scope_location).type.name
+        ].alias()
+        outer_vertex_table = self._current_alias.alias()
+        inner_vertex_type = self._ir.query_metadata_table \
+            .get_location_info(fold_scope_location) \
+            .type \
+            .name
+        outer_vertex_type = self._current_classname
+
+        self._aliases[
+            (self._current_location.query_path, self._fold_inner_vertex_location)
+        ] = inner_vertex_table
+
+        # collect edge information to join the fold subquery to the main selectable
+        edge_direction = fold_scope_location.fold_path[0][0]
+        edge_name = fold_scope_location.fold_path[0][1]
+        full_edge_name = '{}_{}'.format(edge_direction, edge_name)
+
+        edge = self._sql_schema_info.join_descriptors[
+            self._current_classname][full_edge_name]
+
+        # we will use this to make sure that the join inside
+        # the subquery uses one alias of the current table
+        # and that the join outside the subquery uses another alias
+        join_info = (edge,
+                     outer_vertex_table,
+                     inner_vertex_type,
+                     self._current_alias)
+
+        self._fold = SQLFoldObject(inner_vertex_table,
+                                   outer_vertex_table,
+                                   inner_vertex_type,
+                                   outer_vertex_type,
+                                   join_info)
+        # find outputs for this fold in self._all_folded_outputs and add to self._fold_outputs
+        if fold_scope_location.fold_path in self._all_folded_outputs:
+            for fold_output in self._all_folded_outputs[fold_scope_location.fold_path]:
+                if fold_output.field == '_x_count':
+                    raise NotImplementedError(u'_x_count not implemented in SQL')
+
+                intermediate_fold_output_name = self._alias_generator.gen_output_column()
+                self._fold_output_column_aliases[fold_output] = intermediate_fold_output_name
+                # add array aggregated intermediate output to self._fold_outputs
+                self._fold.outputs.append(
+                    sqlalchemy.func.array_agg(
+                        self._fold.inner_vertex_table.c[fold_output.field]
+                    ).label(intermediate_fold_output_name)
+                )
+
+        # the group by field is the unique identifier for the outer vertex
+        self._fold.outputs.append(self._fold.outer_vertex_table.c[edge.from_column])
+        self._fold.group_by.append(self._fold.outer_vertex_table.c[edge.from_column])
+
+    def unfold(self):
+        """Complete the execution of a Fold Block."""
+        # Create the folded select statement, updating the fold count and adding the subquery
+        # to self._aliases
+        # left join to the self._from_clause on self._fold_join_vertex_field
```

Comment:
According to the comments, there's 3 things going on here, and it's not clear what lines of code these comments are talking about. The `_alias` updating is in the next code paragraph. Can you try to organize it into a readable narrative? As I see it, the one thing that happens here is we join the subquery to the main query. Everything else is secondary, as it is maintaining the correctness of our state.